integreat_cms/deepl_api/utils.py

```diff
@@ -0,0 +1,161 @@
+        self.translator = deepl.Translator(settings.DEEPL_AUTH_KEY)
+
+    def check_availability(self, request, language_slug):
+        """
+        This function checks, if the selected language is supported by DeepL
+
+        :param request: request that was sent to the server
+        :type request: ~django.http.HttpRequest
+
+        :param language_slug: current language slug
+        :type language_slug: str
+
+        :return: true or false
+        :rtype: bool
+        """
+        supported_source_languages = [
+            source_language.code.lower()
+            for source_language in self.translator.get_source_languages()
+        ]
+        supported_target_languages = [
+            target_languages.code.lower()[:2]
+            for target_languages in self.translator.get_target_languages()
+        ]
+        return (
+            request.region.default_language.slug in supported_source_languages
+            and language_slug in supported_target_languages
+        )
+
+    def get_translation(self, content, source_language_slug, target_language_key):
+        """
+        This functions gets the translation from DeepL
+
+        :param content: passed content
+        :type content: str
+
+        :param source_language_slug: slug of source language
+        :type source_language_slug: str
+
+        :param target_language_key: slug or in case of pt and en bcp of target language
+        :type target_language_key: str
+
+        :return: translated post in target language
+        :rtype: str
+        """
+        translated_content = self.translator.translate_text(
+            content, source_lang=source_language_slug, target_lang=target_language_key
+        )
+        return translated_content
+
+    def deepl_translation(
+        self, request, content_translations, language_slug, form_class, content_label
+    ):
+        """
+        This functions gets the translation from DeepL
+
+        :param request: passed request
+        :type request: ~django.http.HttpRequest
+
+        :param content_translations: passed content
+        :type content_translations: ~django.db.models.query.QuerySet [ ~integreat_cms.cms.models.abstract_content_translation.AbstractContentTranslation ]
+
+        :param language_slug: current GUI language slug
+        :type language_slug: str
+
+        :param form_class: passed Form class of content type
+        :type form_class: ~integreat_cms.cms.forms.custom_content_model_form.CustomContentModelForm
+
+        :param content_label: passed content type, either event or poi
+        :type content_label: str
+        """
+        # Get target language
+        target_language = request.region.get_language_or_404(language_slug)
+        source_language = request.region.get_source_language(language_slug)
+        for content_translation in content_translations:
+            source_translation = content_translation.get_translation(
+                source_language.slug
+            )
```

Comment:
If the source translation does not exist, I get the following error:
```
AttributeError at /augsburg/events/en/auto-translate/

'NoneType' object has no attribute 'title'
```
So I suggest to show a warning instead.